plugins/by-name/copilot-lua/default.nix

```
@@ -1,172 +1,165 @@
-                      if string.match(vim.fs.basename(vim.api.nvim_buf_get_name(0)), '^%.env.*') then
-                        -- disable for .env files
-                        return false
-                      end
-                      return true
-                    end
-                  \'\';
-                }
-              ```
-
-              The key `"*"` can be used to disable the default configuration.
-              Example:
-              ```nix
-                {
-                  javascript = true; # allow specific filetype
-                  typescript = true; # allow specific filetype
-                  "*" = false; # disable for all other filetypes and ignore default `filetypes`
-                }
-              ```
-            '';
-
-        copilotNodeCommand = mkOption {
-          type = types.str;
-          default = "${pkgs.nodejs-18_x}/bin/node";
-          description = ''
-            Use this field to provide the path to a specific node version such as one installed by
-            `nvm`.
-            Node.js version must be 16.x or newer.
-          '';
+      filetypes = defaultNullOpts.mkAttrsOf' {
+        type = types.bool;
+        pluginDefault = {
+          yaml = true;
+          markdown = false;
+          help = false;
+          gitcommit = false;
+          gitrebase = false;
+          hgcommit = false;
+          svn = false;
+          cvs = false;
+          "." = false;
         };
+        example = lib.literalExpression ''
+          {
+            markdown = true; # overrides default
+            terraform = false; # disallow specific filetype
+            sh.__raw = '''
+              function()
+                if string.match(vim.fs.basename(vim.api.nvim_buf_get_name(0)), '^%.env.*') then
+                  -- disable for .env files
+                  return false
+                end
+                return true
+              end
+            ''';
+            javascript = true; # allow specific type
+            "*" = false; # disable for all other filetypes and ignore default filetypes
+          }
+        '';
+        description = ''
+          Specify filetypes for attaching copilot.
+          Each value can be either a boolean or a lua function that returns a boolean.
+        '';
+      };
 
-        serverOptsOverrides = helpers.defaultNullOpts.mkAttrsOf' {
-          type = types.anything;
-          pluginDefault = { };
-          description = ''
-            Override copilot lsp client `settings`.
-            The settings field is where you can set the values of the options defined in
-            https://github.com/zbirenbaum/copilot.lua/blob/master/SettingsOpts.md.
-            These options are specific to the copilot lsp and can be used to customize its behavior.
+      copilot_node_command = lib.mkOption {
+        type = types.str;
+        default = lib.getExe pkgs.nodejs-18_x;
+        example = lib.literalExpression "lib.getExe pkgs.nodejs";
```

This needs a `defaultText`, otherwise we'll end up evaluating that package to render the docs. And the result won't look great anyway.

```suggestion
        default = lib.getExe pkgs.nodejs-18_x;
        defaultText = lib.literalExpression "lib.getExe pkgs.nodejs-18_x";
        example = lib.literalExpression "lib.getExe pkgs.nodejs";
```